### PR TITLE
Add Register Servlet

### PR DESCRIPTION
**Pull request by mynguyenbmc**
Added RegisterServlet and RegisterServletTest to handle requests to /register URL.
Updated web.xml.
All tests pass and server shows expected result.

More details at: https://sites.google.com/codeustudents.com/main/projects/project-1-adding-a-registration-page/3-add-a-servlet?authuser=1

### CODE REVIEW

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package codeu.controller;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+*Servlet class responsible for user registration.
```
Comment (ElleTojaroon) [inline src/main/java/codeu/controller/RegisterServlet.java]: nit: increase the indent by 2 spaces as per this style guide: https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation

Comment (mynguyenbmc) [reply src/main/java/codeu/controller/RegisterServlet.java]: I tried to fix it and realized that I actually did double indentation in my original files (they looked just like the Google-provided files like LoginServlet.java,etc...)? I'm quite confused

Comment (krtejeda) [reply src/main/java/codeu/controller/RegisterServlet.java]: I think she's just referring to the actual comment, not the code. So like:
/**
 *Servlet class responsible for user registration.
 */
(Just match up the stars) I think the original code was indented just fine.

Comment (krtejeda) [reply src/main/java/codeu/controller/RegisterServlet.java]: My example didn't work...but yeah just match up each new line star with the first star in the first row.

Comment (mynguyenbmc) [reply src/main/java/codeu/controller/RegisterServlet.java]: @krtejeda got it now, thanks!

Comment (mynguyenbmc) [reply src/main/java/codeu/controller/RegisterServlet.java]: @krtejeda @ElleTojaroon fixed it again, let's do a final review so we can merge :)

Comment (mynguyenbmc) [reply src/main/java/codeu/controller/RegisterServlet.java]: @ElleTojaroon can you update the change request? thanks!

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package codeu.controller;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ *Servlet class responsible for user registration.
```
Comment (ElleTojaroon) [inline src/main/java/codeu/controller/RegisterServlet.java]: nit: Put a space or two between * and Servlet.

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.mockito.Mockito;
+
+public class RegisterServletTest {
+
+  private RegisterServlet registerServlet;
+  private HttpServletRequest mockRequest;
+  private PrintWriter mockPrintWriter;
+  private HttpServletResponse mockResponse;
+
+
```
Comment (ElleTojaroon) [inline src/test/java/codeu/controller/RegisterServletTest.java]: nit: remove one empty line so there's one empty line left.

Comment (mynguyenbmc) [reply src/test/java/codeu/controller/RegisterServletTest.java]: just did, thanks :)!

**Pull request merged**